File: bullseye/src/main/resources/bullseye/solutions/MyBullseyePlayer.java

import bullseye.wrappers.java.BullseyePlayer;

import java.util.Random;

/**
 * Bullseye player in Java
 *
 * @author José Carlos Paiva <code>josepaiva94@gmail.com</code>
 */
public class MyBullseyePlayer extends BullseyePlayer {

    Random random = null;
    int lastHorizAngle;
    int lastVertAngle;

    @Override
    public String getName() {
        return "José Carlos Paiva";
    }

    @Override
    public void init() {
        random = new Random();
    }

    @Override
    public void execute() {

        int[] lastHit = getLastHit();

        if (lastHit == null) {
            lastHorizAngle = (int) (random.nextDouble() * 30 - 30);
            lastVertAngle = (int) (random.nextDouble() * 30 - 30);
            shoot(lastHorizAngle, lastVertAngle);

            return;
        }

        if (lastHit[0] > 400) {
            lastHorizAngle = Math.max(lastHorizAngle - 10, -80);
        } else if (lastHit[0] < 200) {
            lastHorizAngle = Math.min(lastHorizAngle + 10, 80);
        } else
            lastHorizAngle = (int) Math.max(Math.min(lastHorizAngle + random.nextDouble() * 2.5, 80), -80);

        if (lastHit[1] > 400) {
            lastVertAngle = Math.max(lastVertAngle - 10, -80);
        } else if (lastHit[1] < 200) {
            lastVertAngle = Math.min(lastVertAngle + 10, 80);
        } else
            lastVertAngle = (int) Math.max(Math.min(lastVertAngle + random.nextDouble() * 2.5, 80), -80);

        shoot(lastHorizAngle, lastVertAngle);
    }
}

File: bullseye/src/main/resources/bullseye/solutions/MyBullseyePlayer.js

// ------------------------ Skeleton --------------------------------

var BullseyePlayer = require('./bullseye/wrappers/js/BullseyePlayer');

/**
 * Bullseye player in Javascript
 *
 * @author José Carlos Paiva <code>josepaiva94@gmail.com</code>
 */
function MyBullseyePlayer() {
    BullseyePlayer.call(this);
}

MyBullseyePlayer.prototype = Object.create(BullseyePlayer.prototype);

// ---------------------- Your code below ---------------------------

/**
 * Get the name of the player
 *
 * @returns {string} name of the player
 */
MyBullseyePlayer.prototype.getName = function () {
    return 'José C. Paiva';
};

/**
 * Initialize the player
 */
MyBullseyePlayer.prototype.init = function () {
};

/**
 * Execute player action
 */
MyBullseyePlayer.prototype.execute = function () {

    var lastHit = this.getLastHit();

    if (lastHit === undefined) {
        this.lastHorizAngle = Math.random() * 30 - 30;
        this.lastVertAngle = Math.random() * 30 - 30;
        this.shoot(this.lastHorizAngle, this.lastVertAngle);

        return;
    }

    if (lastHit[0] > 400) {
        this.lastHorizAngle = Math.max(this.lastHorizAngle - 10, -80);
    } else if (lastHit[0] < 200) {
        this.lastHorizAngle = Math.min(this.lastHorizAngle + 10, 80);
    } else
        this.lastHorizAngle = Math.max(Math.min(this.lastHorizAngle + Math.random() * 2.5, 80), -80);

    if (lastHit[1] > 400) {
        this.lastVertAngle = Math.max(this.lastVertAngle - 10, -80);
    } else if (lastHit[1] < 200) {
        this.lastVertAngle = Math.min(this.lastVertAngle + 10, 80);
    } else
        this.lastVertAngle = Math.max(Math.min(this.lastVertAngle + Math.random() * 2.5, 80), -80);

    this.shoot(this.lastHorizAngle, this.lastVertAngle);
};

// ------------------------ Skeleton --------------------------------

// export this player
module.exports = MyBullseyePlayer;
